src/core/achievements.cpp

```diff
@@ -2059,6 +2061,170 @@ std::string Achievements::GetGameBadgePath(std::string_view badge_name)
   return GetLocalImagePath(badge_name, RC_IMAGE_TYPE_GAME);
 }
 
+bool Achievements::DownloadGameIcons(ProgressCallback* progress, Error* error)
+{
+  progress->SetStatusText("Collecting games...");
+
+  // Collect all unique game IDs that don't have icons yet
+  std::vector<u32> game_ids;
+  {
+    const auto lock = GameList::GetLock();
+    for (const GameList::Entry& entry : GameList::GetEntries())
+    {
+      if (entry.achievements_game_id != 0)
+      {
+        // Check if we already have this badge
+        const std::string existing_badge = GameList::GetAchievementGameBadgePath(entry.achievements_game_id);
+        if (existing_badge.empty() &&
+            std::find(game_ids.begin(), game_ids.end(), entry.achievements_game_id) == game_ids.end())
+        {
+          game_ids.push_back(entry.achievements_game_id);
+        }
+      }
+    }
+  }
+
+  if (game_ids.empty())
+  {
+    progress->SetStatusText("No games need icon downloads.");
+    return true;
+  }
+
+  INFO_LOG("Downloading icons for {} games from RetroAchievements", game_ids.size());
+  progress->SetStatusText(fmt::format("Fetching icon info for {} games...", game_ids.size()).c_str());
```

Comment:
Should use `FormatStatusText()`. The format string itself should be translatable like the others (see `TRANSLATE_FS`).

This also takes a `string_view`, so no `c_str()` is needed.